src/azul/time.py

```diff
@@ -126,17 +126,22 @@ def format_dcp2_datetime(d: datetime) -> str:
     >>> format_dcp2_datetime(datetime(1, 1, 1, tzinfo=timezone.utc))
     '0001-01-01T00:00:00.000000Z'
 
+    >>> format_dcp2_datetime(None)
+
     >>> format_dcp2_datetime(datetime(2020, 1, 1))
     Traceback (most recent call last):
     ...
     azul.RequirementError: 2020-01-01 00:00:00
     """
-    require(str(d.tzinfo) == 'UTC', d)
-    date_string = datetime.strftime(d, dcp2_datetime_format)
-    # Work around https://bugs.python.org/issue13305
-    date_string = ('0000' + date_string)[-31:]
-    assert date_string.endswith('+0000'), date_string
-    return date_string[:-5] + 'Z'
+    if d is None:
+        return None
+    else:
```

Comment:
A fixpoint like this is a smell. If None is a fixpoint in one function why is it not a fixpoint of every function? 

Also, format and parse functions should be each other's inverse and this breaks that.